src/Microsoft.TemplateEngine.TemplateLocalizer/Commands/Export/ExportCommandArgs.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.TemplateLocalizer.Commands.Export
+{
+    /// <summary>
+    /// Model class representing the arguments of <see cref="ExportCommand"/>.
+    /// </summary>
+    internal sealed class ExportCommandArgs
+    {
+        public ExportCommandArgs(IEnumerable<TemplateJsonProvider> templatePath, IEnumerable<string>? language, bool recursive, bool dryRun)
+        {
+            TemplateJsonProviders = templatePath;
+            Languages = language;
+            SearchSubdirectories = recursive;
+            DryRun = dryRun;
+        }
+
+        /// <summary>
+        /// Gets or sets the template.json file providers.
+        /// </summary>
+        public IEnumerable<TemplateJsonProvider> TemplateJsonProviders { get; set; }
```

Comment:
seeing how these properties are initialized in teh .ctor, if you don't expect the values to change, consider making these readonly.

Reply:
Constructor is here to help command line parser map the members to user input. This type itself is basically a model and it should be OK to edit the members if needed (could be useful for testing).

Reply:
you're conscientiously voting down immutability (thread safety) for testability.  Specifically, testing of cases where an instance is created & after instance state is changed via set.

If the type will be relevant to concurrent scenarios, immutability may win in this case.

Reply:
There is no thread safety concern at the moment. I have removed the setters anyway though. Let's start without them and bring them back if needed.